### PR TITLE
issue/6331 - Query class, Meta query, Date queries, Caching

### PR DESCRIPTION
**Pull request by sunnyratilal**
**WIP - Do not merge**

Issue: #6331

### CODE REVIEW

**Comment by JJJ** [comment]
Updated title to reflect changes to #6331.

File: includes/database/queries/class-edd-db-column.php [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+
+	/**
+	 * Return array of allowed column types.
+	 *
+	 * @since 3.0.0
+	 * @access private
+	 * @return array
+	 */
+	private function get_allowed_types() {
+		return (array) apply_filters( 'get_allowed_types', array(
```
Comment (mindctrl) [inline includes/database/queries/class-edd-db-column.php]: If we keep this filter and the one below, we need to prefix them with `edd_<something>`.

File: includes/database/queries/class-edd-db-column.php [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+
+	/**
+	 * Type of database column
+	 *
+	 * @since 3.0.0
+	 * @access public
+	 * @var string
+	 */
+	public $type = '';
+
```
Comment (mindctrl) [inline includes/database/queries/class-edd-db-column.php]: What's the reason for these properties being `public`? Can we make them `protected` or `private` and use a method to retrieve them instead?

File: includes/database/queries/class-edd-db-order-query.php [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+		),
+
+		// total
+		array(
+			'name'       => 'total',
+			'type'       => 'double',
+			'default'    => '0'
+		)
+	);
+
```
Comment (mindctrl) [inline includes/database/queries/class-edd-db-order-query.php]: Same question as above regarding `public`.

File: includes/database/queries/class-edd-db-column.php (hunk 1 above)
Comment (JJJ) [reply includes/database/queries/class-edd-db-column.php]: I think I'll probably remove them for EDD3, and work on them later if there's a business need. We can just use the appropriate MySQL column types until then, and worry about this bit of sanity checking once we identify if this is even a problem.

File: includes/database/queries/class-edd-db-column.php (hunk 2 above)
Comment (JJJ) [reply includes/database/queries/class-edd-db-column.php]: The reason is it's easier for me to architect these bits fully public, and then protect or privatize them later once it's all working.

Protected is probably best IMO. I think we can add methods to get private vars once we start enforcing the shape/sanitization/validation of them.

Comment (JJJ) [reply includes/database/queries/class-edd-db-column.php]: Just tested setting these as `protected` and remembered that even with magic getters, WordPress's list utility functions and methods use `array_key_exists()` internally, so these will need to either:

* Remain explicitly public
* Not be defined at all, which will default them to public when they are set

I think being explicit here is better, even if it's exposing these vars a bit.

File: includes/database/queries/class-edd-db-column.php (hunk 1 above)
Comment (JJJ) [reply includes/database/queries/class-edd-db-column.php]: Opted to remove this filter entirely.

**Comment by JJJ** [comment]
Deleted the `issue/6331` branch.

**Pull request merged**